draco/core/containers.py

```diff
@@ -2566,6 +2567,302 @@ def freq(self):
+        },
+        "kpara": {
+            "axes": ["delay"],
+            "dtype": np.float64,
+            "initialise": True,
+            "distributed": False,
+        },
+        "uv_mask": {
+            "axes": ["u", "v"],
+            "dtype": bool,
+            "initialise": True,
+            "distributed": False,
+        },
+    }
+
+    @property
+    def data(self):
+        """Get the spatial data cube."""
+        return self.datasets["data"]
+
+    @property
+    def kx(self):
+        """Get the kx axis."""
+        return self.datasets["kx"]
+
+    @property
+    def ky(self):
+        """Get the ky axis."""
+        return self.datasets["ky"]
+
+    @property
+    def kpara(self):
+        """Get the k_parallel axis."""
+        return self.datasets["kpara"]
+
+    @property
+    def uv_mask(self):
+        """Get the uv-domain mask."""
+        return self.datasets["uv_mask"]
+
+    @property
+    def redshift(self):
+        """Get the redshift attrs."""
+        return self.attrs["redshift"]
+
+    @property
+    def freq_center(self):
+        """Get the central frequency attrs."""
+        return self.attrs["freq_center"]
+
+
+class PowerSpectrum3D(SpatialDelayCube):
+    """Container for a 3D power spectrum."""
+
+    _dataset_spec: ClassVar = {
+        "spectrum": {
+            "axes": ["pol", "delay", "u", "v"],
+            "dtype": np.float64,
+            "initialise": True,
+            "distributed": True,
+            "distributed_axis": "delay",
+        }
+    }
+
+    @property
+    def spectrum(self):
+        """Get the 3D power spectrum."""
+        return self.datasets["spectrum"]
+
+    @property
+    def ps_norm(self):
+        """Get the power spectrum normalizaiton attrs."""
+        return self.attrs["ps_norm"]
+
+
+class PowerSpectrum2D(CosmologyContainer):
+    """Container for a 2D cylindrically averaged  power spectrum."""
+
+    _axes = ("pol", "delay", "kperp")
+
+    _dataset_spec: ClassVar = {
+        "spectrum": {
+            "axes": ["pol", "delay", "kperp"],
+            "dtype": np.float64,
+            "initialise": True,
+            "distributed": True,
+            "distributed_axis": "delay",
+        },
+        "weight": {
+            "axes": ["pol", "delay", "kperp"],
+            "dtype": np.float64,
+            "initialise": True,
+            "distributed": True,
+        },
+        "neff": {
+            "axes": ["pol", "delay", "kperp"],
+            "dtype": np.float64,
+            "initialise": True,
+            "distributed": True,
+            "distributed_axis": "delay",
+        },
+        "mask": {
+            "axes": ["pol", "delay", "kperp"],
+            "dtype": bool,
+            "initialise": True,
+            "distributed": True,
+        },
+        "kpara": {
+            "axes": ["delay"],
+            "dtype": np.float64,
+            "initialise": True,
+            "distributed": False,
+        },
+        "kperp": {
```

Comment:
I don't think you need the `kperp` dataset, since you are making `kperp` an axis in this container.  Alternatively, you could make the axis something like `uv`, `uv_dist`, or `uv_radius` for consistency with the 3D spectrum and then keep `kperp` as a dataset.